libs/coin-modules/coin-sui/src/network/sdk.ts

```diff
@@ -30,15 +32,32 @@ const BLOCK_HEIGHT = 5; // sui has no block height metainfo, we use it simulate
 
 export const DEFAULT_COIN_TYPE = "0x2::sui::SUI";
 
+type GenericInput<T> = T extends (...args: infer K) => unknown ? K : never;
+type Inputs = GenericInput<typeof fetch>;
+
+const fetcher = (url: Inputs[0], options: Inputs[1], retry = 3): Promise<Response> => {
+  if (options) {
+    options.headers = {
+      ...options.headers,
+      "X-Ledger-Client-Version": getEnv("LEDGER_CLIENT_VERSION"),
+    };
+  }
+  if (retry === 1) return fetch(url, options);
+
+  return fetch(url, options).catch(() => fetcher(url, options, retry - 1));
+};
+
 /**
  * Connects to Sui Api
  */
 export async function withApi<T>(execute: AsyncApiFunction<T>) {
   const url = coinConfig.getCoinConfig().node.url;
+  const transport = new SuiHTTPTransport({
+    url,
+    fetch: fetcher,
+  });
 
-  if (!apiMap[url]) {
-    apiMap[url] = new SuiClient({ url });
-  }
+  apiMap[url] ??= new SuiClient({ transport });
```

Comment:
You could keep the `if (!apiMap[url]) {` to avoid recreating a `SuiHTTPTransport` on every call